### ISSUE TITLE
Need to ensure that servlet-3.1 can't be enabled with ee8 features

### ISSUE THREAD

**Issue by pnicolucci** [OpenLiberty/open-liberty#1350]
Historically the `servlet-3.1` feature has not indicated any `javaeeCompatible` level, meaning that it can be enabled with any version of other EE features (e.g. `jsp-2.2` from EE6).  This probably wasn't intentional, and going forward we want to cap this leniency to make `servlet-3.1` only work with EE6 and EE7 features, and not EE8 and above features.

**Comment by aguibert** [comment]
@pnicolucci and @NottyCode I ran into a potential show-stopper with features that may pull in conflicting EE levels.  For example, if we have:
```xml
    <featureManager>
        <feature>websocket-1.1</feature> <!-- pulls in servlet-3.1 -->
        <feature>collectivecontroller-1.0</feature>  <!-- pulls in restconnector-1.0 (enables jaxrs-1.1) and tolerates rc-2.0 (does not enable jaxrs) -->
    </featureManager>
```

So, if we make this change it makes servlet-3.1 aware of javee compatibility, and can cause transitive dependencies to get pushed around a bit (which is mostly fine), but in this case it would cause a situation where an app could be running today and be accessing jaxrs 1.1 API, and after this change it would break.

so previously we would get these features:
```
[restConnector-1.0, servlet-3.1, ssl-1.0, jndi-1.0, collectiveMember-1.0, collectiveController-1.0, json-1.0, distributedMap-1.0, websocket-1.1, jaxrs-1.1, ].
```
and after the servlet-3.1 alignment with EE6/7 we would get:
```
[restConnector-2.0, servlet-3.1, ssl-1.0, jndi-1.0, collectiveMember-1.0, collectiveController-1.0, json-1.0, distributedMap-1.0, websocket-1.1].
```
so the options I see here are:

1. make a breaking change where some apps may need to enable new features (in this case jaxrs-2.0)
2. let servlet-3.1 be mixed with EE8 features

I'm guessing the answer is (B), but wanted to confirm.

**Comment by pnicolucci** [comment]
Is there anything we can do to ensure we don't have to continue to support servlet-3.1 with future EE features to avoid this issue in the future?

**Comment by aguibert** [comment]
@NottyCode has suggested that we flip the toleration (include eeCompat-6 and tolerate 7).  Giving that a try in PR #1415

**Comment by aguibert** [comment]
Resolved by PR #1415
